src/extra/widgets/chart/lv_chart.c

```diff
@@ -589,7 +585,7 @@ void lv_chart_set_value_by_id2(lv_obj_t * obj, lv_chart_series_t * ser, uint16_t
     if(id >= chart->point_cnt) return;
     ser->x_points[id] = x_value;
     ser->y_points[id] = y_value;
-    lv_chart_refresh(obj);
+    invalidate_point(obj, id);
```

Comment:
I know they both do the same, but I think It's better to keep it like this. As it is more consistent and also maybe someone works on improving the `invalidate_point` for `LV_CHART_TYPE_SCATTER` too.